model/annotations/task_annotations.go

```diff
@@ -213,3 +213,20 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 	)
 	return errors.Wrapf(err, "problem adding task annotation for '%s'", a.TaskId)
 }
+
+func AddCreatedTicket(taskId string, execution int, ticket IssueLink, userDisplayName string) error {
+	source := &Source{
+		Author:    userDisplayName,
+		Time:      time.Now(),
+		Requester: WebHookRequester,
+	}
+	ticket.Source = source
+	_, err := db.Upsert(
+		Collection,
+		ByTaskIdAndExecution(taskId, execution),
+		bson.M{
+			"$addToSet": bson.M{CreatedIssuesKey: ticket},
```

Comment:
I would argue for using $push: the value of addToSet is that only unique items are added, however because Time is always set to time.Now(), multiple API calls for the same ticket would still add the ticket multiple times. $push has the benefit of adding items in order, so they'd be arranged by time in the DB.

Reply:
Thanks for catching that!